src/python/pants/option/options_test.py

```diff
@@ -451,26 +459,34 @@ class Subsystem1(Subsystem):
     def register(opts: Options) -> None:
         opts.register(Subsystem1.options_scope, "--foo")
 
-    opts = create_options(
+    with create_options(
         [GLOBAL_SCOPE],
         register,
         [],
         extra_scope_infos=[Subsystem1.get_scope_info()],
         config={"DEFAULT": {"foo": "aa"}, Subsystem1.options_scope: {"foo": "xx"}},
-    )
-    caplog.clear()
-    assert opts.for_scope(Subsystem1.options_scope).foo == "xx"
-    assert not caplog.records
+    ) as opts:
+        caplog.clear()
+        assert opts.for_scope(Subsystem1.options_scope).foo == "xx"
+        assert not caplog.records
 
 
 def _create_config(
     config: dict[str, dict[str, str]] | None = None,
     config2: dict[str, dict[str, str]] | None = None,
 ) -> Config:
+    config_content = toml.dumps(config or {}).encode()
+    config2_content = toml.dumps(config2 or {}).encode()
+    # Write to files, for the Rust parser.
+    with open("test_config.toml", "wb") as fp:
```

Comment:
I imagine the working directory for running tests is just the default (repository/sandbox root), so this might be junking up the repository if run outside a sandbox (e.g. within an IDE via `pants export ..` etc). Is that expected?

Maybe the first can be managed in the same was as `create_options` with the explicit temporary directory

Reply:
Yeah, this one is complicated, because it's directly used in 8 places, but indirectly (via _parse) in about 70 places...

To fix this I may have to bite the bullet and make the Rust config parser able to parse from memory.

Reply:
I think I'd prefer to do that in a followup though?

Reply:
I can remove the lines that write the files in this PR, since it's not actually needed in main yet, and I can then fix more thoroughly in the branch that integrates the rust parser.